### PR TITLE
Register feature flag page hooks

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Register feature flag page hooks
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| How to test?      | Maybe install PS 1.7.8 and perform a 1.7.7 -> 1.7.8 upgrade to validate it works good ✔️  ?
| Possible impacts? | I modify hook configuration files to acknowledge hooks related to Feature Flags

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24022)

### CODE REVIEW

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 1]
```diff
@@ -70,3 +70,7 @@ UPDATE `PREFIX_product` SET `product_type` = "pack" WHERE `cache_is_pack` = 1;
 UPDATE `PREFIX_product` SET `product_type` = "virtual" WHERE `is_virtual` = 1;
 
 /* PHP:ps_1780_add_feature_flag_tab(); */;
+
+INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `position`) VALUES
+(NULL,"actionFeatureFlagForm","Modify feature flag options form content","This hook allows to modify feature flag options form FormBuilder",1),
+(NULL,"actionFeatureFlagSave","Modify feature flag options form saved data","This hook allows to modify data of feature flag options form after it was saved",1);
```
Comment (Progi1984) [inline install-dev/upgrade/sql/1.7.8.0.sql]: Please move it to l.24

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Please merge sql hooks query

**Review by matthieu-rolland — DISMISSED** [review]

File: install-dev/data/xml/hook.xml [hunk 2]
```diff
@@ -2870,5 +2870,15 @@
       <title>Display after Grid table</title>
       <description>This hook adds new blocks after Grid component table</description>
     </hook>
+    <hook id="actionFeatureFlagForm">
+      <name>actionFeatureFlagForm</name>
+      <title>Modify feature flag page form content</title>
+      <description>This hook allows to modify feature flag page form FormBuilder</description>
```
Comment (eternoendless) [inline install-dev/data/xml/hook.xml]:
```suggestion
      <description>This hook allows to modify the Feature Flag page form's FormBuilder</description>
```

File: install-dev/data/xml/hook.xml [hunk 3]
```diff
@@ -2870,5 +2870,15 @@
...
+    </hook>
+    <hook id="actionFeatureFlagSave">
+      <name>actionFeatureFlagSave</name>
+      <title>Modify feature flag form submitted data</title>
+      <description>This hook allows to modify feature flag data being submitted through the form after it was saved</description>
```
Comment (eternoendless) [inline install-dev/data/xml/hook.xml]:
```suggestion
      <description>This hook allows to modify the Feature Flag data being submitted through the form after it was saved</description>
```

File: install-dev/upgrade/sql/1.7.8.0.sql (hunk 1 above)
Comment (atomiix) [inline install-dev/upgrade/sql/1.7.8.0.sql]:
```suggestion
(NULL, "actionFeatureFlagForm", "Modify feature flag options form content", "This hook allows to modify feature flag options form FormBuilder", 1),
(NULL, "actionFeatureFlagSave", "Modify feature flag options form saved data", "This hook allows to modify data of feature flag options form after it was saved", 1);
```

**Comment by matks** [comment]
> Please merge sql hooks query

Euuuuuh 😅  ? I'm afraid I dont understand.

I'll be happy merging something but I dont see what you mean by SQL hooks query. Is that another PR?

**Comment by atomiix** [comment]
@matks I think what @Progi1984 meant was to group the new sql queries with the already present in the 1.7.8.0.sql file, line 24 😅

**Review by atomiix — DISMISSED** [review]

**Comment by matks** [comment]
> @matks I think what @Progi1984 meant was to group the new sql queries with the already present in the 1.7.8.0.sql file, line 24 😅

Aaaaaah 😛  the word "merge" has a very specific meaning for maintainers 😆  when you tell me to merge something I think about Pull Requests

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 4]
```diff
@@ -21,7 +21,9 @@ INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `po
   (NULL, 'actionFrontControllerSetVariables', 'Add variables in JavaScript object and Smarty templates', 'Add variables to javascript object that is available in Front Office. These are also available in smarty templates in modules.your_module_name.', '1'),
   (NULL, 'displayAdminGridTableBefore', 'Display before Grid table', 'This hook adds new blocks before Grid component table.', '1'),
   (NULL, 'displayAdminGridTableAfter', 'Display after Grid table', 'This hook adds new blocks after Grid component table.', '1'),
-  (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1')
+  (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1'),
+  (NULL, 'actionFeatureFlagForm', 'Modify feature flag options form content', 'This hook allows to modify feature flag options form FormBuilder', 1),
+  (NULL, 'actionFeatureFlagSave', 'Modify feature flag options form saved data', 'This hook allows to modify data of feature flag options form after it was saved', 1);
```
Comment (atomiix) [inline install-dev/upgrade/sql/1.7.8.0.sql]:
```suggestion
  (NULL, 'actionFeatureFlagSave', 'Modify feature flag options form saved data', 'This hook allows to modify data of feature flag options form after it was saved', 1)
```

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 5]
```diff
@@ -21,7 +21,9 @@ INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `po
   (NULL, 'actionFrontControllerSetVariables', 'Add variables in JavaScript object and Smarty templates', 'Add variables to javascript object that is available in Front Office. These are also available in smarty templates in modules.your_module_name.', '1'),
   (NULL, 'displayAdminGridTableBefore', 'Display before Grid table', 'This hook adds new blocks before Grid component table.', '1'),
   (NULL, 'displayAdminGridTableAfter', 'Display after Grid table', 'This hook adds new blocks after Grid component table.', '1'),
-  (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1')
+  (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1'),
+  (NULL, 'actionFeatureFlagForm', 'Modify feature flag options form content', 'This hook allows to modify the Feature Flag page form FormBuilder', 1),
+  (NULL, 'actionFeatureFlagSave', 'Modify feature flag options form saved data', 'This hook allows to modify the Feature Flag data being submitted through the form after it was saved', 1);
```
Comment (PululuK) [inline install-dev/upgrade/sql/1.7.8.0.sql]:
```suggestion
  (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1'),
  (NULL, 'actionFeatureFlagForm', 'Modify feature flag options form content', 'This hook allows to modify the Feature Flag page form FormBuilder', 1),
  (NULL, 'actionFeatureFlagSave', 'Modify feature flag options form saved data', 'This hook allows to modify the Feature Flag data being submitted through the form after it was saved', 1)
```

Comment (PululuK) [reply install-dev/upgrade/sql/1.7.8.0.sql]: oups .. @atomiix already catch ;)

**Review by atomiix — APPROVED** [review]

**Review by PululuK — APPROVED** [review]

**Comment by matks** [comment]
@atomiix @PululuK @eternoendless @Progi1984 thank you for the very quick review !

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]
Tested on a fresh 1.7.8.x install and after an upgrade.
It's QA :heavy_check_mark: 
Thanks @matks

**Comment by PululuK** [comment]
Thanks @matks @boubkerbribri

**Pull request merged**